fml/platform/win/file_win.cc

```diff
@@ -383,12 +383,8 @@ bool WriteAtomically(const fml::UniqueFD& base_directory,
   }
 
   auto file_path = GetAbsolutePath(base_directory, file_name);
-  std::stringstream stream;
-  stream << file_path << ".temp";
-  auto temp_file_path = stream.str();
-
   auto temp_file =
-      OpenFile(temp_file_path.c_str(), true, FilePermission::kReadWrite);
+      OpenFile(file_path.c_str(), true, FilePermission::kReadWrite);
```

Comment:
Adding this back I wasn't able to make ReplaceFile work. Really nothing in this method is atomic for windows though ...

Reply:
I mean this entire method, if anything fails we just leave a temp file laying around. Atomic-ish?

Reply:
That said, I'm gonna say we don't need atomic writes for a shader compiler

Reply:
The atomic part is making sure the final file is entirely what you intended it to be - we want that because otherwise if it's missing some bytes it could cause us to crash at runtime.

Leaving around temp files is bad but it doesn't reduce the atomicity :)

Reply:
the point of what though, that never worked. This is a PR to make impellerc work on windows. If you want to fix WriteFileAtomically for windows that is certainly worthwhile at some point but I don't think its a good use of my time

Reply:
(talked offline - I misunderstood that this was not used yet - I'm fine if we just file a bug with a TODO here to make this be atomic on Windows)

Reply:
https://github.com/flutter/flutter/issues/102838